nebula-logger/plugins/async-failure-additions/plugin/classes/LogBatchApexErrorEventHandler_Tests.cls

```diff
@@ -0,0 +1,102 @@
+//------------------------------------------------------------------------------------------------//
+// This file is part of the Nebula Logger project, released under the MIT License.                //
+// See LICENSE file or go to https://github.com/jongpie/NebulaLogger for full license details.    //
+//------------------------------------------------------------------------------------------------//
+@SuppressWarnings('PMD.ApexDoc, PMD.ApexAssertionsShouldIncludeMessage, PMD.MethodNamingConventions, PMD.ApexUnitTestClassShouldHaveAsserts')
+@IsTest(IsParallel=true)
+private class LogBatchApexErrorEventHandler_Tests implements Database.Batchable<SObject>, Database.RaisesPlatformEvents {
+    private enum Phase {
+        START,
+        EXECUTE,
+        FINISH
+    }
+
+    private final Phase throwLocation;
+
+    @IsTest
+    static void it_should_create_log_when_batch_job_throws_in_start_method() {
+        runTestForPhase(Phase.START);
+    }
+
+    @IsTest
+    static void it_should_create_log_when_batch_job_throws_in_execute_method() {
+        // need an account for the query to actually hand records off to the "execute" method
+        insert new Account(Name = LogBatchApexErrorEventHandler_Tests.class.getName());
+        runTestForPhase(Phase.EXECUTE);
+    }
+
+    @IsTest
+    static void it_should_create_log_when_batch_job_throws_in_finish_method() {
+        runTestForPhase(Phase.FINISH);
+    }
+
+    @SuppressWarnings('PMD.EmptyCatchBlock')
+    private static void runTestForPhase(Phase phase) {
+        Logger.getUserSettings().IsApexSystemDebugLoggingEnabled__c = false;
+        LoggerParameter__mdt mockParam = new LoggerParameter__mdt();
+        mockParam.Value__c = LogBatchApexErrorEventHandler_Tests.class.getName();
+        mockParam.DeveloperName = LogBatchApexErrorEventHandler.BATCH_ERROR_LOGGER + 'Test';
+        LoggerParameter.setMock(mockParam);
+        try {
+            System.Test.startTest();
+            Database.executeBatch(new LogBatchApexErrorEventHandler_Tests(phase));
+            System.Test.stopTest();
+        } catch (Exception ex) {
+            // via https://salesforce.stackexchange.com/questions/263419/testing-batchapexerrorevent-trigger
+        }
+        // at this point, we're still two async-levels deep into Platform Event-land; we need to call "deliver()" twice
+        System.Test.getEventBus().deliver(); // fires the platform event for Database.RaisesPlatformEvents
+        System.Test.getEventBus().deliver(); // fires the logger's platform event
+
+        assertLogWasCreatedForPhase(phase);
+    }
+
+    private static void assertLogWasCreatedForPhase(Phase phase) {
+        Log__c logEntry = getLog();
+        System.assertNotEquals(null, logEntry, 'Log should have been created!');
+        System.assertEquals(2, logEntry.LogEntries__r.size(), 'Two log entries should have been created');
```

Comment:
It looks like this count of 2 is failing in the pipeline because the `extra-tests` folder includes some additional metadata on `Account` that adds more log entries, so it ends up being a higher count here. Thoughts on instead looping through the records in `LogEntry__c` and checking that your specific log entries exist? Or if you have another approach in mind, I'm open to ideas, let me know what you think!